data-prepper-plugins/elasticsearch/src/test/java/com/amazon/dataprpper/plugins/sink/elasticssarch/aws/interceptor/http/AwsRequestSigningApacheInterceptorTest.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2012-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+ * the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package com.amazon.dataprpper.plugins.sink.elasticssarch.aws.interceptor.http;
+
+
+import org.apache.http.HttpEntityEnclosingRequest;
+import org.apache.http.HttpHost;
+import org.apache.http.HttpRequest;
+import org.apache.http.ProtocolVersion;
+import org.apache.http.RequestLine;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.message.BasicHttpEntityEnclosingRequest;
+import org.apache.http.message.BasicHttpRequest;
+import org.apache.http.protocol.BasicHttpContext;
+import org.apache.http.protocol.HttpCoreContext;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import software.amazon.awssdk.auth.credentials.AnonymousCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.core.interceptor.ExecutionAttributes;
+import software.amazon.awssdk.core.signer.Signer;
+import software.amazon.awssdk.http.ContentStreamProvider;
+import software.amazon.awssdk.http.SdkHttpFullRequest;
+import software.amazon.awssdk.http.SdkHttpMethod;
+import software.amazon.awssdk.regions.Region;
+
+import java.io.IOException;
+
+public class AwsRequestSigningApacheInterceptorTest {
```

Comment:
I did not see any tests which do not use `HttpEntityEnclosingRequest`, aside from the Bad Request test. We should probably have at least one test with an `HttpRequest` which is not also a `HttpEntityEnclosingRequest`.

Reply:
Why `HTTPRequest`? `HTTPEntityEnclosingRequest` is an extension of `HTTPRequest`.

Reply:
There is a conditional based on this. While it looks good to visual inspection, we should be sure that we are testing what happens when it is not that type (ie. the conditional is not reached).

Reply:
Its going to be hard to test that because if we send a basichttprequest we will not have content and we want include the content in the signing. I am going to skip this for now.

Reply:
If we don't have content then should be reject the request ?

Reply:
I don't know about rejecting. Would a `GET` or `DELETE` possibly fall under these?